app/models/observation.rb

```diff
@@ -653,73 +655,69 @@ def users_vote(naming, user)
+  # Note: multiple namings can return true for a given observation.
+  # This is used to display eyes next to Proposed Name on Observation page
   def is_owners_favorite?(naming)
     lookup_naming(naming).is_users_favorite?(user)
   end
 
-  # Returns true if a given naming has received the highest positive vote from
-  # the given user (among namings for this observation).  Note, multiple
-  # namings can return true for a given user and observation.
+  # Returns true if a given Naming has received one of the highest positive
+  # votes from the given user (among namings for this observation).
+  # Note: multiple namings can return true for a given user and observation.
   def is_users_favorite?(naming, user)
     lookup_naming(naming).is_users_favorite?(user)
   end
 
+  # rubocop:enable Naming/PredicateName
+
+  # All of observation.user's votes on all Namings for this Observation
+  # Used in Observation and in tests
   def owners_votes
-    users_votes(user)
+    user_votes(user)
   end
 
-  def users_votes(user)
-    result = []
-    namings.each do |n|
+  # All of a given User's votes on all Namings for this Observation
+  def user_votes(user)
+    namings.each_with_object([]) do |n, votes|
       v = n.users_vote(user)
-      result << v if v
+      votes << v if v
     end
-    result
   end
 
-  def owner_favorite_or_explanation
-    if showable_owner_id?
-      owners_only_favorite_name.format_name
-    else
-      :show_observation_no_clear_preference
-    end
+  # Observation.user's unique preferred positive Name for this observation
+  # Returns falsy if there's no unique preferred positive id
+  # Used on show_observation page
+  def owner_preference
+    owner_uniq_favorite_name if owner_preference?
+  end
+
+  private # private methods used by owner_preference ###########################
+
+  # Does observation.user have a single preferred id for this observation?
+  def owner_preference?
+    owner_uniq_favorite_vote&.value&.>= Vote.owner_id_min_confidence
   end
 
-  def owners_only_favorite_name
+  def owner_uniq_favorite_name
     favs = owner_favorite_votes
     favs[0].naming.name if favs.count == 1
   end
 
-  def owner_favorite_vote
-    owner_favorite_votes.first
+  def owner_uniq_favorite_vote
+    votes = owner_favorite_votes
+    return votes.first if votes.count == 1
   end
 
   def owner_favorite_votes
     votes.where(user_id: user_id, favorite: true)
   end
 
-  # show Observer ID? (observer's identification of Observation)
-  # (in code, Observer ID is "owner_id")
-  def show_owner_id?
-    User.view_owner_id_on?
-  end
-
-  def showable_owner_id?
-    owner_sure_enough? && owner_id_known?
-  end
-
-  def owner_sure_enough?
-    return unless owner_favorite_vote
-
-    owner_favorite_vote.value >= Vote.owner_id_min_confidence
-  end
-
-  def owner_id_known?
-    owners_only_favorite_name.try(:known?)
-  end
+  public #######################################################################
```

Comment:
Not a big deal, but in my experience it is unusual to explicitly use `public`.  Generally all the `private` methods just go at the end of the class.  Part of the problem is that this class is really too big so that makes it hard to logically order things.

Reply:
Thanks for he feedback. It also bothered me to have private methods in the middle of the file.
I wanted to keep those methods relatively close to `owner_preference`.
But I'm happy to move them wherever you like.

Reply:
@mo-nathan: 
I will:
- move all the preferred ID stuff to its own section following the Namings and Votes section,
- keep the private methods close to own_preference, rather than 250 lines away. 

Of course, that leaves a `public` in the middle of the file. That's not a strong preference, so if you'd rather have the private stuff at the end, just say the word.